posthog/api/insight.py

```diff
@@ -145,35 +151,54 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Insight:
         validated_data.pop("last_refresh", None)  # last_refresh sometimes gets sent if dashboard_item is duplicated
         tags = validated_data.pop("tags", None)  # tags are created separately as global tag relationships
 
-        if not validated_data.get("dashboard", None):
-            dashboard_item = Insight.objects.create(
-                team=team, created_by=request.user, last_modified_by=request.user, **validated_data
-            )
-        elif validated_data["dashboard"].team == team:
-            created_by = validated_data.pop("created_by", request.user)
-            dashboard_item = Insight.objects.create(
-                team=team, last_refresh=now(), created_by=created_by, last_modified_by=created_by, **validated_data
-            )
-        else:
-            raise serializers.ValidationError("Dashboard not found")
+        created_by = validated_data.pop("created_by", request.user)
+        dashboards = validated_data.pop("dashboards", None)
+
+        insight = Insight.objects.create(
+            team=team, created_by=created_by, last_modified_by=request.user, **validated_data
+        )
+
+        if dashboards is not None:
+            for dashboard in Dashboard.objects.filter(id__in=[d.id for d in dashboards]).all():
+                if dashboard.team != insight.team:
+                    raise serializers.ValidationError("Dashboard not found")
+                DashboardTile.objects.create(insight=insight, dashboard=dashboard)
```

Comment:
Should we add `color` and `layouts` here as well?

Reply:
I think they're both default values at this point so we don't need them...